gwnr/waveform/prepare_waveforms.py

```diff
@@ -0,0 +1,356 @@
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+
+
+Script to 
+
+    a) join h5 files
+    b) extrapolate to infinity
+    c) correct for CoM drift
+    
+    from SXS simulations
+    
+'''
+
+import os
+import scri
+import numpy as np
+from pathlib import Path
+
+
+class PrepareSXSWaveform:
+    ''' Prepare waveforms from a particular Lev and ECC of 
+    an SXS NR runs by
+    
+    1) Joining all segments
+    2) Extrapolating to infinity
+    3) Transforming to CoM frame 
+    4) Upload to cloud 
+    
+    
+    Attributes
+    ----------
+    sim_dir : str/POSIXPath
+              The root directory containing 
+              the simulation directory
+    sim_name : 
+    out_dir
+    joined_outfile_dir
+    joined_waveform_outfile_name
+    joined_waveform_outfile_path
+    joined_horizons_outfile_name
+    joined_waveform_outfile_path
+    lev
+    ecc
+    extrap_out_dir
+    
+    
+    
+    Methods
+    -------
+    join_waveform_h5_files
+    extrapolate
+    join_horizons
+    transform_to_CoM_frame
+    upload_output_dir
+    prepare_waveform
+    
+    
+    
+    '''
+    
+    def __init__(self,
+                 sim_name,
+                 sim_dir=Path('./'),
+                 out_dir=None,
+                 joined_waveform_outfile_name=None,
+                 joined_horizons_outfile_name=None,
+                 lev=2,
+                 ecc=0):
+        
+        
+        if not os.path.isabs(sim_dir):
+            raise ValueError("Please provide the full sim path!")
+            
+        self._sim_dir = Path(sim_dir)
+        
+        self._sim_name = sim_name
+        self._lev = lev
+        self._ecc = ecc
+        
+        if joined_waveform_outfile_name is None:
+            print("Choosing default directory for output...")
+            joined_waveform_outfile_name = sim_name + \
+                                        f"Lev{self.lev}JoinedWaveform.h5"
+            
+        self._joined_waveform_outfile_name = joined_waveform_outfile_name
+        
+        if out_dir is None:
+            self._out_dir = os.path.join(os.getcwd(), 
+                                Path(f'{sim_name}_waveforms_Lev{self.lev}'))
+            #print(f"Creating out directory ({sim_name}_waveforms) in cwd...")
+            
+        else:
+            print(f"Out directory is set to {self.out_dir}")
+            
+        if not os.path.isdir(self.out_dir):
```

Comment:
If a file exists by that name, `isdir` will return `False` and you'll try to create a directory by the same name, throwing an error in the next line.  Remove this `if`.